tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -291,6 +293,34 @@ TangoErrorType TangoRosNode::OnTangoServiceConnected() {
   return TANGO_SUCCESS;
 }
 
+void TangoRosNode::RequestADFPermission() {
+  tango_ros_messages::RequestPermission srv;
+  srv.request.permission = tango_ros_messages::RequestPermission::Request::ADF_PERMISSION;
+  if (request_permission_service_.call(srv)) {
+    if (srv.response.granted) {
+      LOG(INFO) << "ADF permission has been requested and granted by the user.";
+    } else {
+      LOG(WARNING) << "ADF permission has been requested but not granted by the user.";
+    }
+  } else {
+    LOG(ERROR) << "Failed to call service to request ADF permission";
+  }
+}
+
+void TangoRosNode::RequestDatasetPermission() {
+  tango_ros_messages::RequestPermission srv;
+  srv.request.permission = tango_ros_messages::RequestPermission::Request::DATASET_PERMISSION;
+  if (request_permission_service_.call(srv)) {
+    if (srv.response.granted) {
+      LOG(INFO) << "Dataset permission has been requested and granted by the user.";
+    } else {
+      LOG(WARNING) << "Dataset permission has been requested but not granted by the user.";
```

Comment:
I think it will indeed fail to connect if we try to load/record dataset while dataset premission is not granted. I would like to test it but I did not find a proper way to revoke the dataset permission.